### ISSUE TITLE
About the gan-loss

### ISSUE THREAD

**Issue by cannonli7** [facebookresearch/mae#181]
Thanks for your work. Im a little curious about the code of the gan loss in MAE? But i cannot find the code in this repository. could you help me with that

**Comment by tangky22** [comment]
> Thanks for your work. Im a little curious about the code of the gan loss in MAE? But i cannot find the code in this repository. could you help me with that

Hello, I also curious about the gan loss in MAE.  I found the process is the same when loading two models in demo.py and I also couldn't find definiton of gan loss in the code. Have you solve the problem?

**Comment by kevin-Abbring** [comment]
> Thanks for your work. Im a little curious about the code of the gan loss in MAE? But i cannot find the code in this repository. could you help me with that

    def forward_loss(self, imgs, pred, mask):
        """
        imgs: [N, 3, H, W]
        pred: [N, L, p*p*3]
        mask: [N, L], 0 is keep, 1 is remove, 
        """
        target = self.patchify(imgs)
        if self.norm_pix_loss:
            mean = target.mean(dim=-1, keepdim=True)
            var = target.var(dim=-1, keepdim=True)
            target = (target - mean) / (var + 1.e-6)**.5

        loss = (pred - target) ** 2
        loss = loss.mean(dim=-1)  # [N, L], mean loss per patch

        loss = (loss * mask).sum() / mask.sum()  # mean loss on removed patches
        return loss
        
        如果不是，就是？我也没有找到明文的gan_loss代码，只能暂且这么认为。
